torch/distributed/fsdp/_runtime_utils.py

```diff
@@ -154,60 +162,83 @@ def _lazy_init(
+            HandleShardingStrategy._HYBRID_SHARD_ZERO2,
+        ):
+            has_hybrid_sharding_strategy = True
+            # Share the all-reduce state across FSDP units. This is not strictly necessary
             # as each one already uses the same process group, but can slightly save memory
             # since other FSDP units allreduce state can be garbage collected.
             assert inter_node_state is not None, (
                 "`_validate_and_get_hybrid_shard_state()` should have returned "
                 "a valid inter-node state if there exists an FSDP instance "
                 "using a hybrid sharding strategy"
             )
-            fsdp_module._inter_node_state = inter_node_state
+            fsdp_state._inter_node_state = inter_node_state
 
         # Relax the assert for non-root FSDP instances in case the nested
         # initialized module is wrapped again in FSDP later (e.g. after
         # training to run inference)
         p_assert(
-            fsdp_module._is_root is None or not fsdp_module._is_root,
+            fsdp_state._is_root is None or not fsdp_state._is_root,
             "Non-root FSDP instance's `_is_root` should not have been "
             "set yet or should have been set to `False`",
         )
-        fsdp_module._is_root = False
-        fsdp_module._streams = state._streams
-        fsdp_module._stream_to_name = state._stream_to_name
-        fsdp_module._exec_order_data = state._exec_order_data
-        if fsdp_module.limit_all_gathers != state.limit_all_gathers:
-            # Prefer the root's value
-            inconsistent_limit_all_gathers = True
-            fsdp_module.limit_all_gathers = state.limit_all_gathers
-        fsdp_module._free_event_queue = state._free_event_queue
-        fsdp_module._handles_prefetched = state._handles_prefetched
-        fsdp_module._needs_pre_backward_unshard = state._needs_pre_backward_unshard
-        for handle in fsdp_module._handles:
+        fsdp_state._is_root = False
+        fsdp_state._streams = root_state._streams
+        fsdp_state._stream_to_name = root_state._stream_to_name
+        fsdp_state._exec_order_data = root_state._exec_order_data
+        fsdp_state._free_event_queue = root_state._free_event_queue
+        fsdp_state._handles_prefetched = root_state._handles_prefetched
+        fsdp_state._needs_pre_backward_unshard = root_state._needs_pre_backward_unshard
+        for handle in fsdp_state._handles:
             handle.init_flat_param_attributes()
-    if inconsistent_limit_all_gathers:
-        warnings.warn(
-            "Found inconsistent `limit_all_gathers` values across FSDP "
-            f"instances on rank {state.rank}. Using the root FSDP's value of "
-            f"{state.limit_all_gathers} for all instances."
-        )
-    return state
+    for attr_name, attr_values in attr_name_to_values.items():
+        if len(attr_values) != 1:
+            raise ValueError(
```

Comment:
I assume this is for the manual wrapping case, as this will never happen with auto wrap policy?

Reply:
Correct!